src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -38,7 +33,12 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
         /// </summary>
         public string? ClientName { get; }
 
-        public RequestContext(Document? document, Solution solution, Action<Solution>? solutionUpdater, ClientCapabilities clientCapabilities, string? clientName)
+        /// <summary>
+        /// The document that the request is for, if applicable. This comes from the <see cref="TextDocumentIdentifier"/> returned from the handler itself via a call to <see cref="IRequestHandler{RequestType, ResponseType}.GetTextDocumentIdentifier(RequestType)"/>.
+        /// </summary>
+        public Document? Document { get; }
```

Comment:
should this be readonly?

Reply:
Are you thinking of a scenario where it would need to change?

For mutation the intention is that a new solution is supplied to the `UpdateSolution` method, than just assigning a new document.

Reply:
I just meant the fields could be marked with readonly - we create a new request context for each request right?

Reply:
Ohhh yes thank you, either I did this already and accidentally reverted, or just forgot, but making these fields definitely was supposed to be on the list.